org.eclipse.scanning.malcolm.core/src/org/eclipse/scanning/malcolm/core/MalcolmDevice.java

```diff
@@ -47,60 +48,64 @@
 import org.eclipse.scanning.api.points.IPointGenerator;
 import org.eclipse.scanning.api.points.IPosition;
 import org.eclipse.scanning.api.points.models.CompoundModel;
+import org.eclipse.scanning.api.scan.ScanningException;
 import org.eclipse.scanning.sequencer.SubscanModerator;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * Object that make the connection to the device and monitors its status.
  *
- * Important things to do:
- * 1. The locking in AbstractMalcolmDevice should in theory not be needed, we will push this all into the connection.
- * 2. The Serializer to JSON, 'ObjectMapper mapper' must be abstacted out because real connection can be JSON or EPICSV4
- * 3. The Socket to ZeroMQ must be abstracted out because the real socket can be ZeroMQ or EPICSV4
- *
  * @author Matthew Gerring
- *
  */
 public class MalcolmDevice<M extends MalcolmModel> extends AbstractMalcolmDevice<M> {
 
-    // Static data
-	private static Logger logger = LoggerFactory.getLogger(MalcolmDevice.class);
-	private static String STATE_ENDPOINT = "state";
-	private static String HEALTH_ENDPOINT = "health";
-	private static String BUSY_ENDPOINT = "busy";
-	private static String CURRENT_STEP_ENDPOINT = "completedSteps";
-	private static String FILE_EXTENSION_H5 = "h5";
+	private static final Logger logger = LoggerFactory.getLogger(MalcolmDevice.class);
+
+	// Constants
+	private static final String STATE_ENDPOINT = "state";
+	private static final String HEALTH_ENDPOINT = "health";
+	private static final String CURRENT_STEP_ENDPOINT = "completedSteps";
+	private static final String FILE_EXTENSION_H5 = "h5";
+	private static final String STANDARD_MALCOLM_ERROR_STR = "Error from Malcolm Device Connection: ";
 
 	// Frequencies and Timeouts
 	// broadcast every 250 milliseconds
-	private final static long POSITION_COMPLETE_FREQ = Long.getLong("org.eclipse.scanning.malcolm.core.positionCompleteFrequency", 250);
+	private static final long POSITION_COMPLETE_FREQ = Long.getLong("org.eclipse.scanning.malcolm.core.positionCompleteFrequency", 250);
 
-	// Standard timeout for Malcolm Calls
-	private final long getTimeout() {
-		return Long.getLong("org.eclipse.scanning.malcolm.core.timeout",          5*1000);        // 5s
-	}
-	private final long getConfigTimeout() {
-		return Long.getLong("org.eclipse.scanning.malcolm.core.configureTimeout", 10*60*1000);    // 10 min
-	}
-	private final long getRunTimeout() {
-		return Long.getLong("org.eclipse.scanning.malcolm.core.runTimeout",       48*60*60*1000); // 2d
+	private static enum Timeout {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this redundant "static" qualifier. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2786)